### PR TITLE
Turn off S+ switch to non serviceable countries

### PR DESCRIPTION
**Pull request by charleycampbell**
## What does this change?
Due to VAT compliance reasons we cannot support certain countries from having the Supporter Plus product.
As part of this PR BillingCountry is checked and compared with the non-serviceable countries list. This PR changes behaviour on MMA if the billingCountry is a non serviceable country:
- On Account Overview 'Change to monthly + extras' button is no longer visible
- On /reccuringsupport (accessed via the Mange recurring support button) 'Change amount' button is not visible. The user cannot update their amount but can change their payment method or cance
- If a user tries to access the switch journey via the /switch url they are redirected back to the account overview

## How to test
Create a Recurring supporter sub with a billing account as one of the non-servicable countries (OR if you have the code available you can add the billing country of an existing RC to the list temporarily) and log into MMA. Check that you can't see the switch button on account overview, change amount button on manage recurring contribution and that you can't access /switch

## Images

Account Overview without switch button 
image

Manage Recurring support page without Change amount button
image

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: client/components/mma/accountoverview/AccountOverview.tsx [hunk 1]
```diff
@@ -156,8 +157,14 @@ const AccountOverviewPage = () => {
 			isSpecificProductType(productDetail, PRODUCT_TYPES.digipack),
 		);
 
+	const hasNonServiceableCountry = nonServiceableCountries.includes(
+		allActiveProductDetails.find(isProduct)?.billingCountry as string,
+	);
+
 	const isEligibleToSwitch =
-		!maybeFirstPaymentFailure && !hasDigiSubAndContribution;
+		!maybeFirstPaymentFailure &&
```
Comment (KentGrm) [inline client/components/mma/accountoverview/AccountOverview.tsx:165]: I love the "maybe" in this variable name. I wonder if we could make the flag more deterministic? (Not a required change)

File: client/components/mma/accountoverview/ManageProduct.tsx [hunk 2]
```diff
@@ -96,6 +97,10 @@ const InnerContent = ({
 	const maybePatronSuffix =
 		productDetail.subscription.readerType === 'Patron' ? ' - Patron' : '';
 
+	const nonServiceableCountry = nonServiceableCountries.includes(
```
Comment (KentGrm) [inline client/components/mma/accountoverview/ManageProduct.tsx:100]: Would it be better as `isNonServiceableCountry` ?

File: client/fixtures/productBuilder/testProducts.ts [hunk 3]
```diff
@@ -93,6 +93,13 @@ export function contributionCancelled() {
 		.getProductDetailObject();
 }
 
+export function nonServicedCountryContributor() {
```
Comment (KentGrm) [inline client/fixtures/productBuilder/testProducts.ts:96]: Good use of test fixtures!

**Review by KentGrm — APPROVED** [review]
Looks good. One query on a variable name - otherwise looks sound

File: client/components/mma/accountoverview/AccountOverview.tsx (hunk 1 above)
Comment (charleycampbell) [reply client/components/mma/accountoverview/AccountOverview.tsx:165]: I love the maybe!

File: client/components/mma/accountoverview/ManageProduct.tsx (hunk 2 above)
Comment (charleycampbell) [reply client/components/mma/accountoverview/ManageProduct.tsx:100]: Good suggestion!

**Review by KentGrm — APPROVED** [review]

File: client/components/mma/switch/SwitchContainer.tsx [hunk 4]
```diff
@@ -173,6 +177,10 @@ function userIsNavigatingBackFromCompletePage(hasCompleted: boolean) {
 	return hasCompleted && !location.pathname.includes('complete');
 }
 
+function hasNonServiceableCountry(productDetail: ProductDetail) {
```
Comment (andrewHEguardian) [inline client/components/mma/switch/SwitchContainer.tsx:180]: would this also be better as `isNonServiceableCountry`? in fact it's the exact same function as above so you could pull it out to somewhere else, like in Utilities

File: client/components/mma/identity/models.ts [hunk 5]
```diff
@@ -598,3 +598,46 @@ export const COUNTRIES: CountryValues[] = [
 	{ iso: 'ZW', name: 'Zimbabwe' },
 	{ iso: 'AX', name: 'Åland Islands' },
 ];
+
+export const nonServiceableCountries: string[] = [
```
Comment (andrewHEguardian) [inline client/components/mma/identity/models.ts:602]: should this list go in a different file/folder? I'm not sure it's Identity related

**Review by andrewHEguardian — DISMISSED** [review]
👍  nicely done

File: client/components/mma/switch/SwitchContainer.tsx (hunk 4 above)
Comment (charleycampbell) [reply client/components/mma/switch/SwitchContainer.tsx]: Moved into ProductUtils as previously discussed

File: client/components/mma/identity/models.ts (hunk 5 above)
Comment (charleycampbell) [reply client/components/mma/identity/models.ts]: Moved into it's own file

File: client/components/mma/accountoverview/ManageProduct.tsx [hunk 6]
```diff
@@ -156,6 +157,7 @@ const InnerContent = ({
 			)}
 
 			{(isAmountOveridable || showSupporterPlusUpdateAmount) &&
+			!isNonServiceableCountry &&
```
Comment (andrewHEguardian) [inline client/components/mma/accountoverview/ManageProduct.tsx:160]: you're checking the truthiness of the function imported, not calling the function on the productDetail

**Review by andrewHEguardian — APPROVED** [review]

**Pull request merged**